### PR TITLE
Install bundler and all project's gems

### PR DESCRIPTION
**Pull request by acarpe**
After writing to heroku support they pointed me that the 3rd party buildpack should install bundler.
I modified the compile script to install both bundler gem and all the other project's gems.

I tried this on my app and it works well.

this solve the issue #2 
A

### CODE REVIEW

**Comment by febeling** [comment]
@acarpe did Heroku say why bundler should be installed? My unterstanding is: this buildpack depends on the two other buildpack (node and ruby), and should have gems stalled by the ruby buildpack already.

**Comment by acarpe** [comment]
They didn't provide a motivation about it.
From what I remember any build pack can decide to expose executables or PATH like the nodejs does, but for some reason the ruby build pack do not expose its `bundle` executable.

The original response from the support was:

# 

Hi there,

Troubleshooting third party buildpacks is outside the scope of Heroku Support. The buildpacks' authors would have to add the bundler gem into their buildpack in order to enable it's use.

It's not necessary to rely on a third party buildpack to enable this functionality, I recommend looking into our Release Phase beta which enables you to run commands like rake tasks after a build completes, but before the app is executed.

Let me know if you have any further questions.

Thanks!

# Jason

File: bin/compile [hunk 1]
```diff
@@ -28,8 +28,8 @@ export_env_dir() {
...
 export_env_dir
 echo "-----> gem install bundler"
-gem install bundler
+gem install bundler --no-ri --no-rdoc
 echo "-----> bundle install"
-bundle install
+bundle install --without development:test --path vendor/bundle --binstubs vendor/bundle/bin -j4 --deployment
 echo "-----> bundle exec rails webpack:compile"
-bundle exec rails webpack:compile --trace
+RAILS_ENV=production bundle exec rails webpack:compile --trace
```
Comment (febeling) [inline bin/compile:35]: @acarpe – im not sure about some of these improvements. Can you explain them individually? 
- `--no-ri`, `--no-rdoc` seem to be options for `gem` and bundler has no such options, but doing what they mean for gem by default
- `--deployment` seems to imply `--path vendor/bundle`
- Do we care about the binstub directory?
- Setting `RAILS_ENV` to `production` would prevent setting it to something else, like a staging env hosted on Heroku?

`--j4`, `--deployment` and `--without development:test` look good to me!

**Comment by febeling** [comment]
I wonder if the bundle executable in path is something that depends on the stack version..?

File: bin/compile [hunk 2]
```diff
@@ -27,6 +27,9 @@ export_env_dir() {
...
 export_env_dir
-
-echo "-----> bundle exec rake webpack:compile"
-bundle exec rake webpack:compile --trace
+echo "-----> gem install bundler"
+gem install bundler
+echo "-----> bundle install"
+bundle install
+echo "-----> bundle exec rails webpack:compile"
+bundle exec rails webpack:compile --trace
```
Comment (febeling) [inline bin/compile]: I don't think the rails command is the right thing to invoke here, is it?

**Comment by febeling** [comment]
Thanks for the PR! I've used your approach while fixing some issues and commited 43b26de and 15417f3.

**Pull request closed without merging**